Library/OcCryptoLib/gost3411-2012-core.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2013, Alexey Degtyarev <alexey@renatasystems.org>.
+ * All rights reserved.
+ *
+ * GOST R 34.11-2012 core and API functions.
+ *
+ * $Id$
+ */
+
+#ifdef OC_CRYPTO_SUPPORTS_STREEBOG
+
+#include "gost3411-2012-core.h"
+
+#define BSWAP64(x) \
+    (((x & 0xFF00000000000000ULL) >> 56) | \
+     ((x & 0x00FF000000000000ULL) >> 40) | \
+     ((x & 0x0000FF0000000000ULL) >> 24) | \
+     ((x & 0x000000FF00000000ULL) >>  8) | \
+     ((x & 0x00000000FF000000ULL) <<  8) | \
+     ((x & 0x0000000000FF0000ULL) << 24) | \
+     ((x & 0x000000000000FF00ULL) << 40) | \
+     ((x & 0x00000000000000FFULL) << 56))
+
+VOID
+GOST34112012Cleanup (
+  GOST34112012Context  *CTX
+  )
+{
+  for (INT32 i = 0; i < 64; ++i) {
+    CTX->buffer[i] = 0;
+  }
+
+  for (INT32 i = 0; i < 8; ++i) {
+    CTX->hash.QWORD[i]  = 0;
+    CTX->h.QWORD[i]     = 0;
+    CTX->N.QWORD[i]     = 0;
+    CTX->Sigma.QWORD[i] = 0;
+  }
+
+  CTX->bufsize     = 0;
+  CTX->digest_size = 0;
+}
+
+VOID
+GOST34112012Init (
+  GOST34112012Context  *CTX,
+  CONST UINT32         digest_size
+  )
+{
+  UINT32  i;
+
+  GOST34112012Cleanup (CTX);
+  CTX->digest_size = digest_size;
+
+  for (i = 0; i < 8; i++) {
+    if (digest_size == 256) {
+      CTX->h.QWORD[i] = 0x0101010101010101ULL;
+    } else {
+      CTX->h.QWORD[i] = 0x00ULL;
+    }
+  }
+}
+
+VOID
+Pad (
```

Comment:
Do not add functions with such generic names into global namespace. Make them `STATIC` or change the name adding Streebog prefix.